### ISSUE TITLE
Figure out how important DEFAULTINIT is in ScalaSignatures

### ISSUE THREAD

**Issue by xeno-by** [twitter/rsc#123]
SemanticDB doesn't have the `DEFAULTINIT` property, so we can't easily emit the `DEFAULTINIT` flag in ScalaSignatures. Is this going to be a problem? /cc @olafurpg

**Comment by xeno-by** [comment]
There is one read usage of `DEFAULTINIT` in 2.11.12:
  * [Symbol.isInitializedToDefault](https://github.com/scala/scala/blob/v2.11.12/src/reflect/scala/reflect/internal/Symbols.scala#L812), only under `-Xcheckinit`, which is itself used only once:
      * [Mixin.needsInitFlag](https://github.com/scala/scala/blob/v2.11.12/src/compiler/scala/tools/nsc/transform/Mixin.scala#L91)

There are two read usages of `DEFAULTINIT` in 2.12.6:
  * [Symbol.isInitializedToDefault](https://github.com/scala/scala/blob/v2.12.6/src/reflect/scala/reflect/internal/Symbols.scala#L818), which is never used.
  * [AccessorSynthesis.computeBitmapInfos](https://github.com/scala/scala/blob/v2.12.6/src/compiler/scala/tools/nsc/transform/AccessorSynthesis.scala#L143), only under `-Xcheckinit`, which is used twice:
      * [Fields](https://github.com/scala/scala/blob/v2.12.6/src/compiler/scala/tools/nsc/transform/Fields.scala#L74)
      * [Mixin](https://github.com/scala/scala/blob/v2.12.6/src/compiler/scala/tools/nsc/transform/Mixin.scala#L16)

If I understand Scalac code correctly, not setting `DEFAULTINIT` will only have effect under `-Xcheckinit` and will simply lead to extra checks (that `var x: T = _` are initialized before use). We don't use `-Xcheckinit` and even in the worst case the effect is pretty benign, so I'm closing this as wontfix.

**Comment by dragos** [comment]
You're right, it's only a problem under `-Xcheckinit`, but the effects would be worse than some extra checks: it changes how bits are assigned in the bitmap for initialized fields and may lead to conflicts between different fields. The bitmap is shared with lazy values, so some lazy vals might look initialized even though they were not. 

That said, I don't think many people use `-Xcheckinit` so this won't be a blocker to adoption of rsc.

**Comment by xeno-by** [comment]
Uh oh. This looks worse that I thought - thank you for bringing this up! 

For our toolchain, that's like super accessors (https://github.com/twitter/rsc/issues/100) - something that's missing in SemanticDB, can't be computed from SemanticDB alone, but is essential to binary compatibility with Scalac. We'll need to deal with this, so I'll reopen.
